Document CMake option VISIT_CONFIG_SITE (#17445)
visit-dav/visit: src/doc/building_visit/Building_Directly_With_CMake.rst

.. _Building Directly with CMake:

Building Directly with CMake
----------------------------

> If the third party libraries have already been installed using ``build_visit``, 

Reviewer: Just a quick FYI before I continue review...

That isn't obvious from most of the *existing* `.rst` content because the bulk of it is still in the *older* style so I wouldn't expect you to know this but...we no longer [split sentences across lines](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/ContributingToDocumentation/index.html#about-line-length).

Author: Ok, thanks for that info. I actually read that section of the docs yesterday, and then I was looking at a bunch of other .rst files and I thought that maybe the docs were out of date! I had it backwards.

> VisIt_ can be built without the use of the ``build_visit`` script, with these 

Reviewer: ```suggestion
If a *config site file* is available for the platform you wish to build on, VisIt_ can often be built without the use of the ``build_visit`` script, with these 
```

steps.

.. code:: bash

  git clone git@github.com:visit-dav/visit.git
  mkdir visit/build
  cd visit/build

> ``build_visit`` should have created a cmake file specific to your machine.

Reviewer: ```suggestion
If ``build_visit`` was used to build VisIt_ on the platform in the past, it should have created a cmake file specific to your machine which we call a *config site* file.
```

> CMake needs to be simply pointed at it. Examples can be found in the 

Reviewer: ```suggestion
CMake simply needs to be told where to find it using the ``-DVISIT_CONFIG_SITE`` option. Examples of *config site* files for a variety of machines VisIt_ developers directly support can be found in the 
```

> ``src/config-site`` directory.

Reviewer: Can you make the ``src/config-site`` text here a clickable link to our repository's config-site dir?

Author: How can I do that? Or alternatively, where can I find a good example of it?

Reviewer: Hmmm... 🤔 maybe scan through some pages of the hosted docs like [this one](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/ContributingToDocumentation/index.html#about-line-length) looking for cases where you see links out of the docs to somewhere. Then scroll to top of those pages in browser and hit the `Edit on GitHub` button in upper right. That'll get you to that file. Then, finally, view the **Raw** file (GitHub tries to interpret `.rst` files so viewing **Raw** is helpful in allowing you to see the actually ascii source).

All that said, there are a few bullets describing how links work in our doc's [Contributing to documentation](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/ContributingToDocumentation/index.html#about-line-length) section.

If after trying any of that, it still isn't obvious, ping again here.
